sinatra-example.rb

```diff
@@ -17,4 +19,77 @@ class SinatraExample < Sinatra::Base
     "slept for #{milliseconds} milliseconds"
   end
 
-end
\ No newline at end of file
+  get '/mem/alloc/:size_mb/?:leak?/?:once?' do
+    content_type 'application/text'
+
+    if params[:once] and params[:once]=1 and $leak_buffer
+      puts "Already has memory allocated, skipping"
+      return "Already has memory allocated, skipping"
+    end
+
+    size_mb = params[:size_mb].to_i
+    total_bytes = size_mb*1024**2
+    do_leak = (params[:leak] and params[:leak] == "1")
+
+    # Metrics before allowcate
+    rss_bytes_before = OS.rss_bytes
+    free_mem_before = (Vmstat.memory.free * Vmstat.memory.pagesize) / 1024
+
+    puts "Allocating: #{size_mb} MB. Leaking memory: #{if do_leak; "Yes" else  "No" end}, system free: #{free_mem_before/1024} MiB, Process rss: #{rss_bytes_before/1024} MiB"
+
+    # Allocate the memory chunk
+    chunk_of_mem = 'a'*(total_bytes-1)
+    if do_leak
+        $leak_buffer = [] unless $leak_buffer
+        $leak_buffer << chunk_of_mem
+    end
+
+    sleep(1.0/10.0) # To be sure that Vmstat will catch the memory change
```

Comment:
How about sleep(0.1) instead? Is there any specific reason why you divide two real numbers here?